docs/templates/base.html

```diff
@@ -0,0 +1,144 @@
+{% extends "adidoks/templates/base.html" %}
+{% block footer %}
+{{ super() }}
+<script>
+document.addEventListener("DOMContentLoaded", inject);
+
+function inject() {
+  document.querySelectorAll(".code-selector").forEach(injectTabs);
+}
+
+function injectTabs(codeSelector) {
+  const nav = codeSelector.querySelector("ul.nav");
+
+  let langs = [];
+  let tabs = [];
+  let currentLang = null;
+  codeSelector.childNodes.forEach((element) => {
+    let toAdd = null;
+
+    if (element.nodeType === Node.COMMENT_NODE) {
+      // if its a comment like <!-- go -->, then set the currentLang to go and add it to langs
+      // end will cancel the currentLang, so you can continue to write common documentation.
+      const lang = element.data.trim();
+      if (lang === "end") {
+        currentLang = null;
+      } else {
+        currentLang = lang;
+        toAdd = { lang, element };
+      }
+    } else if (element.tagName === "PRE") {
+      // pre with data-lang attribute
+      const lang = element.getAttribute("data-lang");
+      // code blocks reset the currentLang.
+      currentLang = null;
+      toAdd = { lang, element };
+    } else if (currentLang) {
+      // if we are in a currentLang, then add the element to it.
+      toAdd = { lang: currentLang, element };
+    }
+
+    if (!toAdd) {
+      return;
+    }
+
+    langs.push(toAdd);
+    if (tabs.indexOf(toAdd.lang) === -1) {
```

Comment:
nitty nit: `!tabs.includes(toAdd.lang)`

Another alternative is to use a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) and convert as needed using `Array.from(mySet)`